mongoDB-case-study.md

```diff
@@ -0,0 +1,113 @@
+
+## Weaknesses
+
+- Uses a lot of memory for data storage.
+- Security may not be their strongest suite as there has been reports of MongoDB installations has been stolen, and servers has been held for ransome. Creators claim to have taken measures to upscale the security.
+- Joining - there isn't a simple, automated way to join documents. This can be done, but has to be done manually
+- Limited data size - a document can't be more than 16MB.
+- Limitation to nesting - it can nest, but there is no nesting above 100 levels.
+
+## MongoDB compared with CouchDB
+
+CouchDB is also a NoSQL document oriented database. The design favors availability and consistency. The documents are JSON based and are stored in databases. The software can be installed as single instance or a cluster of instances, this makes it durable and gives it give availability.
+
+CouchDB and MongoDB compared on some key features:
+
+#### Data Model
+
+- MongoDB: stores the data in BSON format - documents gets grouped in databases and collections and are returned as JSON. They support a large amount of data types, including: strings, numbers, geo data, dates, arrays, decimal, nested objects, binary data.
+- CouchDB: stores JSON documents. They only support data types like: string, numbers, arrays, objects and booleans.
+
+#### Indexing
+
+- MongoDB: indexes are supported in many different types: Compound, Text, Geo, Wild Card, TTL, and Partial.
+- CouchDB: indexing is limited, and for some workloads it requires a "design document" definition.
+
+#### Query Language
+
+- MongoDB: Uses their own query language, MongoDB Query Language (MQL). Designed for easy use for developers.
+- CouchDB: Uses HTTP protocol for API, javascript.
+
+#### Availability and Scalability
+
+- MongoDB: Using replica and sharding mechanisms. Replica in short: Makes copies of the document so when primary fails, secondary will become primary. Makes it fault tolerant. Sharding in short: Makes it easy to scale collections over replica sets, geo-zone sharding makes it easy to manage data.
+- CouchDB: Also uses a concept of replica and sharding. Replica in short: Makes copies of the documents, which makes this fault tolerant, but each "node" is also the shard, the sharding is a automated mechanism making it not possible for the user to define keys or groupings.
+
+This comparison is created with the help of <a href="https://www.mongodb.com/compare/couchdb-vs-mongodb"> MongoDBs own comparison</a>, and <a href="https://couchdb.apache.org/"> CouchDBs homepage </a>.
+
+## Summary
+
+Overall MongoDB is a strong NoSQL database. The strengths trumps over the weaknesses. MongoDB is easy to use, it has many features many developers look for, and it supports many different development languages, which makes it more accessible for different types of developers. It can contain big amounts of data and the data is easy accessed.
```

Comment:
```suggestion
Overall MongoDB is a strong NoSQL database. In the author's view, the strengths trump the weaknesses. MongoDB is easy to use, it has many features many developers look for, and it supports many different development languages, which makes it more accessible for different types of developers. It can contain large amounts of data, and the data is easily accessed.
```